app/services/settings-v2/video.ts

```diff
@@ -3,80 +3,201 @@ import { Inject } from 'services/core/injector';
 import { InitAfter } from 'services/core';
 import { mutation, StatefulService } from '../core/stateful-service';
 import * as obs from '../../../obs-api';
-import { SettingsManagerService } from 'services/settings-manager';
+import { GreenService } from 'services/green';
+
+/**
+ * Display Types
+ *
+ * Add display type options by adding the display name to the displays array
+ * and the context name to the context name map.
+ */
+const displays = ['default', 'horizontal', 'green'] as const;
+export type TDisplayType = typeof displays[number];
 
 export function invalidFps(num: number, den: number) {
   return num / den > 1000 || num / den < 1;
 }
 
-@InitAfter('UserService')
-export class VideoSettingsService extends StatefulService<{ videoContext: obs.IVideo }> {
-  @Inject() settingsManagerService: SettingsManagerService;
+export interface IVideoSetting {
+  default: obs.IVideoInfo;
+  horizontal: obs.IVideoInfo;
+  green: obs.IVideoInfo;
+}
+
+@InitAfter('SceneCollectionsService')
+export class VideoSettingsService extends StatefulService<IVideoSetting> {
+  @Inject() greenService: GreenService;
 
   initialState = {
-    videoContext: null as obs.IVideo,
+    default: null as obs.IVideoInfo,
+    horizontal: null as obs.IVideoInfo,
+    green: null as obs.IVideoInfo,
   };
 
   init() {
     this.establishVideoContext();
+    if (this.greenService.views.activeDisplays.green) {
+      this.establishVideoContext('green');
+    }
   }
 
+  contexts = {
+    default: null as obs.IVideo,
+    horizontal: null as obs.IVideo,
+    green: null as obs.IVideo,
+  };
+
   get videoSettingsValues() {
-    const context = this.state.videoContext;
+    const context = this.contexts.horizontal;
+    return {
+      baseRes: `${context.video.baseWidth}x${context.video.baseHeight}`,
+      outputRes: `${context.video.outputWidth}x${context.video.outputHeight}`,
+      scaleType: context.video.scaleType,
+      fpsType: context.video.fpsType,
+      fpsCom: `${context.video.fpsNum}-${context.video.fpsDen}`,
+      fpsNum: context.video.fpsNum,
+      fpsDen: context.video.fpsDen,
+      fpsInt: context.video.fpsNum,
+    };
+  }
```

Comment:
i think for this one and `get baseResolution` we actually do want to reference state since these values are displayed to the user

Reply:
Done